tests/src/test/scala/com/nvidia/spark/rapids/spill/SpillFrameworkSuite.scala

```diff
@@ -0,0 +1,1103 @@
+      assertResult(true)(handle.spillable)
+
+      withResource(handle.materialize(sparkTypes)) { _ =>
+        // 0 because we have a reference to the host batch
+        assertResult(false)(handle.spillable)
+        assertResult(0)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+      }
+
+      // after closing we still have 0 bytes in the store or available to spill
+      assertResult(true)(handle.spillable)
+    }
+  }
+
+  test("host originated: a host batch is not spillable when columns are incRefCounted") {
+    val (hostCb, sparkTypes) = buildHostBatch()
+    val sizeOnHost = RapidsHostColumnVector.getTotalHostMemoryUsed(hostCb)
+    withResource(SpillableHostColumnarBatchHandle(hostCb)) { handle =>
+      assertResult(true)(handle.spillable)
+      val leakedFirstColumn = withResource(handle.materialize(sparkTypes)) { cb =>
+        // 0 because we have a reference to the host batch
+        assertResult(false)(handle.spillable)
+        assertResult(0)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+        // leak it by increasing the ref count of the underlying cuDF column
+        RapidsHostColumnVector.extractBases(cb).head.incRefCount()
+      }
+      withResource(leakedFirstColumn) { _ =>
+        // 0 because we have a reference to the first column
+        assertResult(false)(handle.spillable)
+        assertResult(0)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+      }
+      // batch is now spillable because we close our reference to the column
+      assertResult(true)(handle.spillable)
+      assertResult(sizeOnHost)(SpillFramework.stores.hostStore.spill(sizeOnHost))
+    }
+  }
+
+  test("host originated: an aliased host batch is not spillable (until closing the original) ") {
+    val (hostBatch, sparkTypes) = buildHostBatch()
+    val handle = SpillableHostColumnarBatchHandle(hostBatch)
+    withResource(handle) { _ =>
+      assertResult(1)(SpillFramework.stores.hostStore.numHandles)
+      assertResult(true)(handle.spillable)
+      withResource(handle.materialize(sparkTypes)) { _ =>
+        assertResult(false)(handle.spillable)
+      } // we now have two copies in the store
+      assertResult(true)(handle.spillable)
+    }
+  }
+
+  test("host originated: a host batch supports duplicated columns") {
+    val (hostBatch, sparkTypes) = buildHostBatchWithDuplicate()
+    val handle = SpillableHostColumnarBatchHandle(hostBatch)
+    withResource(handle) { _ =>
+      assertResult(1)(SpillFramework.stores.hostStore.numHandles)
+      assertResult(true)(handle.spillable)
+      withResource(handle.materialize(sparkTypes)) { _ =>
+        assertResult(false)(handle.spillable)
+      } // we now have two copies in the store
+      assertResult(true)(handle.spillable)
+    }
+  }
+
+  test("host originated: a host batch supports aliasing and duplicated columns") {
+    SpillFramework.shutdown()
+    val sc = new SparkConf
+    // disables the host store limit
```

Comment:
Is this just a confusing config name where setting it (enabled) to true disables it (because it enables some alternative mechanism or something), or is the comment wrong?

Reply:
The way we have defined host limits is that if they are set, the host spill's store own limits are ignored. 

The idea is that host spills will be triggered only by a host OOM, and not via a host store limit. So yes, enabling the off heap limit, disables the host store limit.  I'll try to change the comment.